### PR TITLE
feat(ddm): Allow access to public use case ids and private only in admin

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR adds the support for visibility of use cases in the metrics endpoints. With the new system, we will allow to query only public use cases from the API, unless you are a superuser. In that case you will get access to all of them.

### CODE REVIEW

File: tests/sentry/api/endpoints/test_organization_metrics.py [hunk 1]
```diff
@@ -79,27 +79,43 @@ class OrganizationMetricsPermissionTest(APITestCase):
...
+    def test_access_with_wrong_permission_scopes(self):
         with assume_test_silo_mode(SiloMode.CONTROL):
             token = ApiToken.objects.create(user=self.user, scope_list=[])
 
         for method, endpoint, *rest in self.endpoints:
-            response = self.send_request(token, method, endpoint, *rest)
+            response = self.send_request(self.organization, token, method, endpoint, *rest)
+            assert response.status_code == 403
+
+    def test_access_of_another_organization(self):
```
Comment (iambriccardo) [inline tests/sentry/api/endpoints/test_organization_metrics.py:102]: Added this test to make sure no user can access data of another organization.

File: tests/sentry/api/endpoints/test_organization_metrics_details.py [hunk 2]
```diff
@@ -13,6 +19,11 @@
 class OrganizationMetricsDetailsTest(OrganizationMetricsIntegrationTestCase):
     endpoint = "sentry-api-0-organization-metrics-details"
 
+    #         self.user = self.create_user(email="a@example.com", is_managed=False, name="example name")
+    #         self.superuser = self.create_user(is_superuser=True)
+    #         self.staff_user = self.create_user(is_staff=True)
+    #         self.login_as(user=self.user)
```
Comment (obostjancic) [inline tests/sentry/api/endpoints/test_organization_metrics_details.py:25]:
```suggestion
```

File: tests/sentry/sentry_metrics/test_gen_metrics_multiprocess_steps.py [hunk 3]
```diff
@@ -424,7 +424,7 @@ def test_process_messages_default_card_rollout(set_sentry_option: Callable[...,
     ),
     (
         {
-            "name": "c:transactions/alert@none" * 21,
+            "name": "c:transactions/alert@" + ("none" * 50),
```
Comment (obostjancic) [inline tests/sentry/sentry_metrics/test_gen_metrics_multiprocess_steps.py:427]: nit: this looks unrelated

File: tests/sentry/sentry_metrics/test_rh_metrics_multiprocess_steps.py [hunk 4]
```diff
@@ -368,7 +368,7 @@ def test_process_messages() -> None:
     ),
     (
         {
-            "name": SessionMRI.RAW_ERROR.value * 21,
```
Comment (obostjancic) [inline tests/sentry/sentry_metrics/test_rh_metrics_multiprocess_steps.py:371]: nit: this looks unrelated

File: src/sentry/api/endpoints/organization_metrics.py [hunk 5]
```diff
@@ -57,46 +62,78 @@
...
-    UseCaseID.SPANS,
-    UseCaseID.CUSTOM,
-    UseCaseID.PROFILES,
-    UseCaseID.METRIC_STATS,
-]
 
+def can_access_use_case_id(request: Request, use_case_id: UseCaseID) -> bool:
+    api_access = get_use_case_id_api_access(use_case_id)
+    return api_access == UseCaseIDAPIAccess.PUBLIC or (
+        has_elevated_mode(request) and api_access == UseCaseIDAPIAccess.PRIVATE
```
Comment (obostjancic) [inline src/sentry/api/endpoints/organization_metrics.py:69]: looking at the comments in `has_elevated_mode` we probably should not rely to that, i think that `is_active_superuser` is what we want here

**Review by obostjancic — APPROVED** [review]
Looks ok, left a couple of comments ;)

File: tests/sentry/sentry_metrics/test_gen_metrics_multiprocess_steps.py (hunk 3 above)
Comment (iambriccardo) [reply tests/sentry/sentry_metrics/test_gen_metrics_multiprocess_steps.py:427]: Oh yeah, it messed up since I rebased the PR. Will figure out why.

File: src/sentry/api/endpoints/organization_metrics.py (hunk 5 above)
Comment (iambriccardo) [reply src/sentry/api/endpoints/organization_metrics.py:69]: I was told by @markstory that this would be the correct way to check it and it will be refactored in the future to a newer function.

Comment (iambriccardo) [reply src/sentry/api/endpoints/organization_metrics.py:69]: If you check the definition of the function it uses `is_active_superuser`. https://github.com/getsentry/sentry/blob/9e8670622e9b2c04dc1eab32b41886db19668d55/src/sentry/auth/elevated_mode.py#L68-L68
